addon/components/osf-paginator/component.js

```diff
@@ -0,0 +1,68 @@
+import Ember from 'ember';
+import layout from './template';
+
+// Ember component that is very similar to osf/website/static/js/paginator.js
+export default Ember.Component.extend({
+    layout,
+    currentPage: 1,
+    pages: Ember.computed('totalSearchResults', function() {
+        let totalSearchResults = this.get('totalSearchResults');
+        return Math.ceil(totalSearchResults / 10);
+    }),
+    paginators: Ember.computed('currentPage', 'pages', function() {
+        let currentPage = this.get('currentPage') - 1;
+        var MAX_PAGES_ON_PAGINATOR = 7;
+        var MAX_PAGES_ON_PAGINATOR_SIDE = 5;
+        var pages = this.get('pages');
+        var paginator = Ember.A();
+
+        if (pages > 1) {
+            paginator.pushObject('<');
+            paginator.pushObject(1);
+        }
+
+        if (pages <= MAX_PAGES_ON_PAGINATOR) {
+            for (var i = 1; i < pages - 1; i++) {
+                paginator.pushObject(i + 1);
+            }
+        } else if (currentPage < MAX_PAGES_ON_PAGINATOR_SIDE - 1) {
+            for (var j = 1; j < MAX_PAGES_ON_PAGINATOR_SIDE; j++) {
+                paginator.pushObject(j + 1);
+            }
+            paginator.pushObject('...');
+        } else if (currentPage > pages - MAX_PAGES_ON_PAGINATOR_SIDE) {
+            paginator.pushObject('...');
+            for (var k = pages - MAX_PAGES_ON_PAGINATOR_SIDE; k < pages - 1; k++) {
+                paginator.pushObject(k + 1);
+            }
+        } else {
+            paginator.pushObject('...');
+            for (var l = currentPage - 1; l <= currentPage + 1; l++) {
+                paginator.pushObject(l + 1);
+            }
+            paginator.pushObject('...');
+        }
+
+        paginator.push(pages);
+        paginator.push('>');
```

Comment:
Am I correct in assuming that most of this code is borrowed from the current OSF paginator implementation?

We'll probably want to rework this in the future, though I do recognize that for now we just need something to move forward with.